View/ViewHandler.php

```diff
@@ -104,17 +106,33 @@ class ViewHandler extends ContainerAware implements ConfigurableViewHandlerInter
      */
     protected $contextAdapter;
 
+    private $router;
+    private $serializer;
+    private $templating;
+    private $requestStack;
+    private $exceptionWrapperHandler;
+
     /**
      * Constructor.
      *
-     * @param array  $formats              the supported formats as keys and if the given formats uses templating is denoted by a true value
-     * @param int    $failedValidationCode The HTTP response status code for a failed validation
-     * @param int    $emptyContentCode     HTTP response status code when the view data is null
-     * @param bool   $serializeNull        Whether or not to serialize null view data
-     * @param array  $forceRedirects       If to force a redirect for the given key format, with value being the status code to use
-     * @param string $defaultEngine        default engine (twig, php ..)
+     * @param RouterInterface                  $router                  The router
+     * @param object                           $serializer              An object implementing a serialize() method
+     * @param EngineInterface                  $templating              The configured templating engine
+     * @param RequestStack                     $requestStack            The request stack
+     * @param ExceptionWrapperHandlerInterface $exceptionWrapperHandler An exception wrapper handler
+     * @param array                            $formats                 the supported formats as keys and if the given formats uses templating is denoted by a true value
+     * @param int                              $failedValidationCode    The HTTP response status code for a failed validation
+     * @param int                              $emptyContentCode        HTTP response status code when the view data is null
+     * @param bool                             $serializeNull           Whether or not to serialize null view data
+     * @param array                            $forceRedirects          If to force a redirect for the given key format, with value being the status code to use
+     * @param string                           $defaultEngine           default engine (twig, php ..)
      */
     public function __construct(
+        RouterInterface $router,
+        $serializer,
```

Comment:
imho its also worth consider using setter injection for those "optional" or non strongly typed dependencies and then using an internal getter that handles the case of a missing dependency